### PR TITLE
Implement new QA builds for both Android and iOS

### PR DESCRIPTION
**Pull request by Fatxx**
**Description**

- This PR implements new QA apps for both Android and iOS;
- It adds three new workflows in Bitrise: `build_android_qa`, `build_ios_qa` and `create_qa_builds`;
- It renames the previous release builds in Bitrise to: `build_android_release`, `build_ios_release` and `create_release_builds`;
- "About MetaMask" screen shows the current branch where the build was triggered from in QA build;

**Screenshots/Recordings**
iOS:
Screen Shot 2022-08-10 at 22 51 05
Screen Shot 2022-08-10 at 22 52 40
Android:
Screen Shot 2022-08-10 at 23 35 11
Screen Shot 2022-08-10 at 23 34 57

**Related documents**

[Notion Knowledge Share](https://www.notion.so/Mobile-App-Targets-Variants-44eda191e09f41a8af93b89684dda22d)

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/344 and https://github.com/MetaMask/mobile-planning/issues/345

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+      - deploy-to-bitrise-io@1:
+          inputs:
+            - deploy_path: $PROJECT_LOCATION/app/build/outputs/bundle/prodRelease/app-prod-release.aab
+          title: Bitrise Deploy AAB
+      - deploy-to-bitrise-io@1:
+          inputs:
+            - deploy_path: sourcemaps/android/index.js.map
+          title: Bitrise Deploy Sourcemaps
+
+  android_internal_release_step:
```
Comment (Fatxx) [inline bitrise.yml:254]: Change to `android_qa_step`

File: bitrise.yml [hunk 2]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+      - deploy-to-bitrise-io@1:
+          inputs:
+            - deploy_path: $PROJECT_LOCATION/app/build/outputs/bundle/qaRelease/app-qa-release.aab
+          title: Bitrise Deploy AAB
+      - deploy-to-bitrise-io@1:
+          inputs:
+            - deploy_path: sourcemaps/android/index.js.map
+          title: Bitrise Deploy Sourcemaps
+
+  android_store_release_step:
```
Comment (Fatxx) [inline bitrise.yml:300]: Change to `run_ios_qa_app`

File: android/app/build.gradle [hunk 3]
```diff
@@ -226,9 +229,30 @@ android {
 			multiDexEnabled true
 			shrinkResources true
 			proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-			signingConfig signingConfigs.release
 		}
 	}
+
+	flavorDimensions "version"
+   		productFlavors {
+        dev {
```
Comment (sethkfman) [inline android/app/build.gradle:237]: I think we can remove this product flavor to stay inline with iOS, that has two targets

File: bitrise.yml [hunk 4]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+          inputs:
+            - deploy_path: $BITRISE_DEPLOY_DIR/app-release.apk
+          title: Bitrise Deploy APK
+      - yarn@0:
+          inputs:
+            - command: build:announce
+          title: Accouncing pre-release
+          is_always_run: false
+
+  android_test_release_step:
```
Comment (sethkfman) [inline bitrise.yml:208]: Delete this step

File: bitrise.yml [hunk 5]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+
+  ios_store_release_step:
         before_run:
             - ios_test_release_step
         steps:
             - deploy-to-itunesconnect-application-loader@1:
                   inputs:
                       - ipa_path: ios/build/output/MetaMask.ipa
+
+  ios_test_release_step:
```
Comment (sethkfman) [inline bitrise.yml:323]: Delete this step

Comment (sethkfman) [inline bitrise.yml:323]: Merge this into `ios_store_release_step`

File: bitrise.yml [hunk 6]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+            - access_token: $BITRISE_START_BUILD_ACCESS_TOKEN
+      - script@1:
+          inputs:
+            - content: |-
+                #!/usr/bin/env bash echo 'weew - everything passed!'
+          title: All Tests Passed
+          is_always_run: false
+
+  # Parallel Build & Deploy Steps
+  start_bitrise_release_builds:
```
Comment (sethkfman) [inline bitrise.yml:163]: Change to `run_qa_app`

File: bitrise.yml [hunk 7]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+            - wait_for_builds: 'true'
+            - abort_on_fail: 'yes'
+            - access_token: $BITRISE_START_BUILD_ACCESS_TOKEN
+      - build-router-wait@0:
+          inputs:
+            - abort_on_fail: 'yes'
+            - build_artifacts_save_path: $BITRISE_DEPLOY_DIR
+            - access_token: $BITRISE_START_BUILD_ACCESS_TOKEN
+
+  start_store_release_builds:
```
Comment (sethkfman) [inline bitrise.yml:181]: Change to `run_prod_app`

File: bitrise.yml (hunk 5 above)
Comment (sethkfman) [inline bitrise.yml:323]: Remove "test" from this step title

File: bitrise.yml (hunk 4 above)
Comment (sethkfman) [inline bitrise.yml:208]: Remove "test" from this step title

File: bitrise.yml [hunk 8]
```diff
@@ -3,295 +3,387 @@ format_version: '8'
...
+            - app_path: $PROJECT_LOCATION/app/build/outputs/bundle/prodRelease/app-prod-release.aab
+            - track: internal
+            - service_account_json_key_path: $BITRISEIO_BITRISEIO_SERVICE_ACCOUNT_JSON_KEY_URL_URL
+            - package_name: $MM_ANDROID_PACKAGE_NAME
+    envs:
+      - opts:
+          is_expand: false
+        MM_ANDROID_PACKAGE_NAME: io.metamask
+
+  ios_store_release_step:
```
Comment (sethkfman) [inline bitrise.yml:315]: Change to `run_ios_prod_app`

File: package.json [hunk 9]
```diff
@@ -354,20 +361,28 @@
...
+        "binaryPath": "android/app/build/outputs/apk/qa/debug/app-qa-debug.apk",
         "build": "yarn start:android:e2e",
         "type": "android.emulator",
         "device": {
           "avdName": "Pixel_3_API_29"
         }
       },
       "android.emu.release": {
-        "binaryPath": "android/app/build/outputs/apk/release/app-release.apk",
+        "binaryPath": "android/app/build/outputs/apk/qa/release/app-prod-release.apk",
```
Comment (sethkfman) [inline package.json:372]: This should be 
```suggestion
        "binaryPath": "android/app/build/outputs/apk/prod/release/app-prod-release.apk",
```

File: package.json [hunk 10]
```diff
@@ -354,20 +361,28 @@
         }
       },
       "android.emu.debug": {
-        "binaryPath": "android/app/build/outputs/apk/debug/app-debug.apk",
+        "binaryPath": "android/app/build/outputs/apk/qa/debug/app-qa-debug.apk",
```
Comment (sethkfman) [inline package.json:364]:
```suggestion
        "binaryPath": "android/app/build/outputs/apk/prod/debug/app-prod-debug.apk",
```

File: scripts/build.sh [hunk 11]
```diff
@@ -151,7 +152,7 @@ prebuild_android(){
 
 buildAndroidRun(){
 	prebuild_android
-	react-native run-android
+	react-native run-android --variant=devDebug
```
Comment (sethkfman) [inline scripts/build.sh:155]:
```suggestion
	react-native run-android --variant=prodDebug
```

File: scripts/build.sh [hunk 12]
```diff
@@ -271,12 +345,15 @@ buildAndroid() {
...
 
 buildIos() {
+  echo "Build iOS $MODE started..."
 	if [ "$MODE" == "release" ] ; then
 		buildIosRelease
 	elif [ "$MODE" == "releaseE2E" ] ; then
 		buildIosReleaseE2E
 	elif [ "$MODE" == "debugE2E" ] ; then
 		buildIosSimulatorE2E
+  elif [ "$MODE" == "QA" ] ; then
```
Comment (sethkfman) [inline scripts/build.sh:355]: format

File: ios/MetaMask/IosExportOptionsStoreRelease.plist [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>provisioningProfiles</key>
+    <dict>
+      <key>io.metamask.MetaMask</key>
+      <string>Bitrise AppStore io.metamask.MetaMask</string>
```
Comment (sethkfman) [inline ios/MetaMask/IosExportOptionsStoreRelease.plist:8]: What about updating the name of the file to `ios/MetaMask/IosExportOptionsProductionRelease.plist`

File: ios/MetaMask/IosExportOptionsInternalRelease.plist [hunk 14]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+  <dict>
+    <key>provisioningProfiles</key>
+    <dict>
+      <key>io.metamask.MetaMask-QA</key>
+      <string>Bitrise Internal Release - MetaMask-QA</string>
```
Comment (sethkfman) [inline ios/MetaMask/IosExportOptionsInternalRelease.plist:8]: What about updating the name of the file to `ios/MetaMask/IosExportOptionsQARelease.plist`

File: package.json [hunk 15]
```diff
@@ -356,20 +363,28 @@
         }
       },
       "android.emu.debug": {
-        "binaryPath": "android/app/build/outputs/apk/debug/app-debug.apk",
+        "binaryPath": "android/app/build/outputs/apk/prod/debug/app-qa-debug.apk",
```
Comment (sethkfman) [inline package.json:366]:
```suggestion
        "binaryPath": "android/app/build/outputs/apk/prod/debug/app-prod-debug.apk",
```

**Review by sethkfman — COMMENTED** [review]
Minor fix on the build.sh but other than that it looks good.

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**